src/tests/JIT/opt/And/IntAnd.cs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace CodeGenTests
+{
+    class IntAnd
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void SideEffect()
+        {
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static bool Test_UInt32_UInt32_And(uint x, uint y)
+        {
+            // X64-NOT: movzx
+
+            // We expect 'and reg8, reg8'.
+            // X64: and {{[a-z]+[l|b]}}, {{[a-z]+[l|b]}}
```

Comment:
For future consideration, not here: since we have a rewriting step for FileCheck anyway, does it make sense to provide some abbreviations for common regexes?

For example, choose a new special character such as ~, then make ~reg8\~ expand to your regex above (and ~~ to quote an actual ~).  Then you'd get
```
// X64: and {{~reg8~}}, {{~reg8~}}
```
and not need the comment.

Reply:
I have thought about this too. I agree that we should do something like this.